src/custom/mirage-gallery-curated/index.js

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import slugify from "slugify";
+
+function getProjectID(tokenId) {
+  let tokenStr = tokenId.toString();
+  if (tokenStr.length === 5) {
+    return parseInt(tokenStr[0], 10);
+  } else if (tokenStr.length === 6) {
+    return parseInt(tokenStr.slice(0, 2), 10);
+  } 
+}
+
+export const fetchCollection = async (_chainId, { contract, tokenId }) => {
+  
+  const projectID = getProjectID(tokenId);
+
+  const url = `https://account.miragegallery.ai/curated-details.json`;
+  const { data } = await axios.get(url);
+
+  const projectDetails = data.data.find(item => item.projectId === projectID);
+  const miragePayoutAddress = data.curatedPayoutAddress;
+  const mirageBPS = data.curatedPayoutBPS;
+  const artistBPS = data.artistPayoutBPS;
+
+  const startTokenId = tokenId - (tokenId % 10000);
+  const endTokenId = startTokenId + 10000 - 1;
+
+  let royalties;
+
+  if (projectDetails.secondaryArtistAddress == "") {
+    royalties = [{bps: mirageBPS, recipient: miragePayoutAddress}, {bps: artistBPS, recipient: projectDetails.artistAddress}]
+  } else {
+    royalties = [{bps: mirageBPS, recipient: miragePayoutAddress}, {bps: artistBPS / 2, recipient: projectDetails.artistAddress}, {bps: artistBPS / 2, recipient: projectDetails.secondaryArtistAddress}]
+  }
+
+  let extURL = projectDetails.website
+  if (projectDetails.website == "") {
+    extURL = "https://miragegallery.ai/curated"
+  }
+
+  return {
+    id: `${contract}:${startTokenId}:${endTokenId}`,
+    slug: slugify(projectDetails.name, { lower: true }),
```

Comment:
this should be the opensea slug, see two comments below for how to retrieve it

Reply:
@nofir I assume this is just in order to properly aggregate collection offers? 

Similar to comment above, I'm hoping to not rely on OpenSea's frontend.

Let me know if this is do-able or not.

Reply:
Yes, we use that for collection offers. The logic to pull the slug and fees out of opensea already exists and just need to be added according to the example i sent. 

This is not a requirement but would mean some limitations on any orders being ingested to our system from opensea

Reply:
@nofir alright I just commit some changes. Let me know if there are any other issues.